containers/src/Kokkos_DualView.hpp

```diff
@@ -552,6 +591,41 @@ template< class , class , class , class > friend class DualView ;
 #endif
   }
 
+  inline void modify_host() {
```

Comment:
@crtrott @mhoemmen should there be a guard to check that `modified_flags.data() != NULL`? In Tpetra's DistObject `doTransferNew` routine a call to `modify_host()` replaced a call to `clear_sync_state()` causing a seg fault in some `TpetraCore_BlockCrsMatrix` tests, chasing some other stuff down now.
If `modified_flags.data() == NULL` then I'm assuming the function does nothing? I can make the change, just confirming the behavior.

Reply:
Is it possible for `modified_flags` to be null?

Reply:
@mhoemmen  Yes, though I can't say if that's the case by intention.

Reply:
@mhoemmen Just for clarity, I'm saying yes based on debugging PR trilinos/Trilinos#3682 and occurrences of `modified_flags` being null resulting in seg fauilts.